docs/features/search/search-for-plugin-devs.md

```diff
@@ -0,0 +1,187 @@
+---
+id: for-plugin-devs
+title: Search for Plugin Developers
```

Comment:
I'm thinking about our docs structure now, we also have the how-to guides section and maybe some parts in there could be moved to this. What about having 

`How to guides - Plugin Devs`
`How to guides - Adopters`

or potentially even keep the high level menu item of "How to guides" and then the two above could become sub pages of that? What do you think?

Reply:
Yeah, as I was thinking through where to put this content, it made me realize that it might be useful to segment _the entire_ docs section by audience.  Quite a bit of the docs we have in the main search section are oriented toward adopters, and this is the one piece of new information that feels kinda out of place.

With that in mind, I see there's an [Integrate into the Software Catalog](https://backstage.io/docs/plugins/integrating-plugin-into-software-catalog) section under the `Plugins` section.  I wonder if this might be more appropriate on that level too?  e.g.

```diff
Plugins
  - Plugin Development
  - Structure of a Plugin
+ - Integrate with Search
  - Integrate into the Software Catalog
```

Whaddya think @emmaindal?

Reply:
Went ahead and moved it over; could be worth a deeper reorganization of audience-specific documentation.